image-search-abstraction-layer/api.js

```diff
@@ -17,42 +16,44 @@ module.exports = function(app) {
     var pages = req.query.page;
     var sizes = req.query.size;
     var client = new googleSearch(process.env.CSEID, process.env.APIKEY);
-    
+
     mongo.connect(process.env.MONGO_URI, function (err, db) {
       if (err) throw err;
       console.log('Connected to mongoDB');
-      
+
       var collection = db.collection('searches');
       var time = new Date();
       time /= 1000;
 
-      collection.insert({'searchQuery': searchQuery, 'timeSearched': moment.unix(time).format('MMMM Do YYYY, h:mm:ss a')});    
+      collection.insert({
+        searchQuery: searchQuery,
+        timeSearched: moment.unix(time).format('MMMM Do YYYY, h:mm:ss a'),
+      });
       db.close();
     });
 
-    client.search(searchQuery, {size: sizes, page: pages}).then(images => {
-      return res.json({images});
-     });
+    client.search(searchQuery, { size: sizes, page: pages }).then((images) => {
+      return res.json({ images });
+    });
   }
 
   function getRecent(req, res) {
     mongo.connect(process.env.MONGO_URI, function (err, db) {
       if (err) throw err;
```

Comment:
This LGTM 👍 

---

Two Questions:
1) Does the connection/db close on `throw`?
2) What problem does this PR exactly fix? That is, generally, I would have expected us to just make use of MongoDB's queryTimeout, instead.

Reply:
Awesome, thanks for taking a look at this @ShaunSHamilton, and for your questions.

Yes, I believe the connection and DB both close automatically on `throw`. I haven't been able to test this in every scenario, but if I change the Mongo password in my `.env` file and search for images or hit the `recent` endpoint, the application crashes. That should automatically close any open connects to the database.

About your second question, sorry, I didn't provide proper context when I first opened this PR.

In the chat, Yoko mentioned that a Japanese contributor reported the same issue as the one listed here: https://github.com/freeCodeCamp/freeCodeCamp/issues/43518

I believe the problem was that, now that there are a number of recent searches, it created a race condition where the connection would be closed before the recent searches could be returned. Moving `db.close()` into the `.toArray()` seems to fix this.

I did try chaining `.then()` to `.toArray()`, and using `async / await`, but that didn't seem to work. I think it would be possible to wrap some of that code in a `Promise`, though I'm not sure how much we want to mess with it.